### DISCUSSION TITLE
tutorials/assets_pipeline/importing_3d_scenes/node_type_customization

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#279] [User-contributed notes]
# tutorials/assets_pipeline/importing_3d_scenes/node_type_customization

Many times, when editing a scene, there are common tasks that need to be done after exporting: Adding collision detection to objects., Setting objects as navigation meshes., Deleting nodes that are...

https://docs.godotengine.org/en/4.3/tutorials/assets_pipeline/importing_3d_scenes/node_type_customization.html

**Comment by Innokentiy-Alaytsev** [comment]
Judging by the [Godot source code](https://github.com/godotengine/godot/blob/a40fc2354a1639f283c7c46ba8dc321b7ff15960/editor/import/3d/resource_importer_scene.cpp#L414) and the observed behaviour, `$*` are processed as import hint suffixes as well, e.g. `$colonly` is the same as `-colonly`. The comment says "blender and other stuff" whatever that means.

**Comment by etherealxx** [comment]
> In Blender, this requires using the NLA Editor and naming the Action with the loop or cycle prefix or suffix.

When using NLA, there are Actions, Strips, and Tracks. Based on my testing, you need to rename the **Tracks**, not the Action. Atleast when i export into glb with "NLA Tracks" export mode.

**Comment by tobia88** [comment]
It's odd because only Collada files support primitive collision shape, neither gltf is supporting.
Collision shape will be imported as a ConcavePolygonShape3D even exported with a simple cube in gltf.
i would suggest supporting more suffixes like -cononly-box, -cononly-sphere , -cononly-capsule...etc to enhance support for exporting primitive shapes.

**Comment by Monday-nr** [comment]
"loop" for animations doesn't really work.
If you open the imported glb file as a scene, the animation will be set to loop in the scene's animation player.
But if you just double click the glb file to open the import settings, the loop mode is not set there. And setting the save path for the animations and reimporting will get rid of the loop data for good (since the suffix is removed from the names already). And the "saved" animation files will not be set to loop. (Godot 4.4.1)

**Reply by Calinou** [reply]
This sounds like a bug, can you open an issue on https://github.com/godotengine/godot/issues with a minimal reproduction project attached?

**Reply by Monday-nr** [reply]
I opened one: https://github.com/godotengine/godot/issues/108823

**Reply by Sasino97** [reply]
I ended up writing a post import plugin for this

**Comment by GreenCloversGames** [comment]
Tried some experiments and you can't combine multiple types of suffixes, so for example you can't combine "-colonly" and "-rigid", you simply get a static body that has the name ending in -rigid. 
It also doesn't seem like it's possible to change the type of body in the import editor without having to create a script to do so. 
So if you're looking to generate a rigid body out of simple shapes, you'll need to do manually, which is a shame. If you know of a better way, please do leave a comment!

**Comment by chaosjoey** [comment]
I would love if tags for navigation link start position and end positions could be added.

**Reply by Calinou** [reply]
Please open a proposal for this on the [Godot proposals](https://github.com/godotengine/godot-proposals) repository.

**Comment by RostagRed** [comment]
Is there not a way to import a collision without it being a Static Body?

**Reply by Calinou** [reply]
You can import it as a RigidBody instead using `-rigid`.

**Reply by Sasino97** [reply]
It would be nice to only import it as a CollisionShape3D instead, useful when the scene root itself is set to a rigid/static/vehicle body.

**Reply by Sasino97** [reply]
I ended up creating a GltfDocumentExtension that replaces the mesh with a CollisionShape3D based on it in the _ImportNode method, if the node ends with the _SHAPE suffix. I guess this could also have been done as a ScenePostImportPlugin. The thing I love about Godot is that it can be extended so easily if something is lacking.

**Comment by RostagRed** [comment]
Nice! Thanks dude.
